workspaces/liferay-sample-workspace/client-extensions/liferay-sample-custom-element-2/src/common/components/Comic.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
```

Comment:
I rewrote this whole file using more modern react code, let me know what you think, it should function the same.


```jsx
import React from 'react';

function Comic({oAuth2Client}) {
	const [comicData, setComicData] = React.useState(null);

	React.useEffect(() => {
		const request = oAuth2Client.fetch('/comic').then((comic) => {
			setComicData({
				alt: comic.alt,
				img: comic.img,
				title: comic.safe_title,
			});
		});

		return () => request.cancel();
	}, []);

	return !comicData ? (
		<div>Loading...</div>
	) : (
		<div>
			<h2>{comicData.title}</h2>

			<p>
				<img alt={comicData.alt} src={comicData.img} />
			</p>
		</div>
	);
}

export default Comic;
```

Reply:
If you like it, I like it!